### PR TITLE
ref #627: order portal nodes in navigation module by main tree node

### PR DESCRIPTION
**Pull request by fabianbarth-esono**
| Q             | A
| ------------- | ---
| Branch        | 6.3.x
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed issues  | chameleon-system/chameleon-system#627
| License       | MIT

### CODE REVIEW

File: src/CoreBundle/Bridge/Chameleon/Module/NavigationTree/NavigationTree.php [hunk 1]
```diff
@@ -572,7 +572,12 @@ private function createRootTree(): array
             $defaultPortal = $this->portalDomainService->getDefaultPortal();
             $defaultPortalMainNodeId = $defaultPortal->fieldMainNodeTree;
 
-            $portalList = \TdbCmsPortalList::GetList();
+            $portalsInMainTreeNodeOrderQuery = '
+                SELECT `cms_portal`.* FROM `cms_portal` 
+                    LEFT JOIN `cms_tree` ON `cms_portal`.`main_node_tree` = `cms_tree`.`id` 
+                ORDER BY `cms_tree`.`lft`, `cms_portal`.`sort_order`
```
Comment (UlrichKu) [inline src/CoreBundle/Bridge/Chameleon/Module/NavigationTree/NavigationTree.php:578]: Why order by lft?

File: src/CoreBundle/Bridge/Chameleon/Module/NavigationTree/NavigationTree.php [hunk 2]
```diff
@@ -572,7 +572,12 @@ private function createRootTree(): array
             $defaultPortal = $this->portalDomainService->getDefaultPortal();
             $defaultPortalMainNodeId = $defaultPortal->fieldMainNodeTree;
 
-            $portalList = \TdbCmsPortalList::GetList();
+            $portalsInMainTreeNodeOrderQuery = '
```
Comment (UlrichKu) [inline src/CoreBundle/Bridge/Chameleon/Module/NavigationTree/NavigationTree.php:575]: -> matchingPortalsQuery ?

The current name doesn't help me much. Probably everyone needs to look at the query itself to really understand ist.

File: src/CoreBundle/Bridge/Chameleon/Module/NavigationTree/NavigationTree.php [hunk 3]
```diff
@@ -572,7 +572,12 @@ private function createRootTree(): array
...
+                SELECT `cms_portal`.* FROM `cms_portal` 
+                    LEFT JOIN `cms_tree` ON `cms_portal`.`main_node_tree` = `cms_tree`.`id`
```
Comment (UlrichKu) [inline src/CoreBundle/Bridge/Chameleon/Module/NavigationTree/NavigationTree.php:577]: What does this filter out?

Comment (fabianbarth-esono) [reply src/CoreBundle/Bridge/Chameleon/Module/NavigationTree/NavigationTree.php:577]: Nothing (just a left join, not an inner join). It just provides the cms_tree fields needed in ORDER BY.

File: src/CoreBundle/Bridge/Chameleon/Module/NavigationTree/NavigationTree.php (hunk 2 above)
Comment (fabianbarth-esono) [reply src/CoreBundle/Bridge/Chameleon/Module/NavigationTree/NavigationTree.php:575]: The query does not match anything else, it just orders the entries. Would $portalsOrderedByMainTreeNodePosition be more helpful?

File: src/CoreBundle/Bridge/Chameleon/Module/NavigationTree/NavigationTree.php (hunk 1 above)
Comment (fabianbarth-esono) [reply src/CoreBundle/Bridge/Chameleon/Module/NavigationTree/NavigationTree.php:578]: The nested set is used to render all the other parts of the navigation tree. In my opinion, it makes sense to use it for the portal-level too.

File: src/CoreBundle/Bridge/Chameleon/Module/NavigationTree/NavigationTree.php (hunk 2 above)
Comment (UlrichKu) [reply src/CoreBundle/Bridge/Chameleon/Module/NavigationTree/NavigationTree.php:575]: orderedPortalsQuery ?

Comment (fabianbarth-esono) [reply src/CoreBundle/Bridge/Chameleon/Module/NavigationTree/NavigationTree.php]: Renamed.

**Pull request merged**